main.go

```diff
@@ -130,21 +173,27 @@ func (p Proxy) ServeHTTP(w http.ResponseWriter, r *http.Request, next caddyhttp.
 		return nil
 	}
 
-	to, err := p.authRedirect(r)
+	upstream, err := p.authRedirect(r)
 	if err != nil {
 		return err
 	}
-	caddyhttp.SetVar(r.Context(), "upstream", to)
-	p.log.Info("setting upstream to " + to)
+
+	if upstream == p.api || upstream.Path == "" {
```

Comment:
I don't quite understand this if-condition. It looks like it's saying, "If we're sending them to the authorization API, or if the upstream server we're sending them to doesn't define a specific URL path, then use whatever URL path they requested."

However, if we're sending them to the authorization API, then we only want them to go to a single, pre-defined URL path... the one that sets the JWT with the authorization data.

Would you mind clarifying this condition's purpose for me? I suspect I'm still missing/misunderstanding something about it.

Reply:
We want to redirect them back to wherever they were before, not just "/", though I'll probably need to adjust it some to use a query param (not sure if I can rewrite with a query param included or not, it didn't look like it when I tried at first, but I may have not been doing it correctly).